ls1intum/Artemis
# Update server dependencies and add open api specification

Medium updates in Spring Boot, Spring Cloud and Spring Messaging which led to some refactoring.
Also some minor client dependency updates
Fixes #1382

### Checklist

### Motivation and Context
We should stay up to date

### Description
Refactored some web socket related code, in particular moved logs and metrics into the MetricsBean to avoid dependency cycles within 

### Steps for Testing
1. Developers should test the development setup, starting the server and client application and communicating using web sockets on the localhost should still work as usual 
2. Testers should click through most views of the Artemis application and review if you notice issues

### Test Coverage
no changes

### Screenshots
no UI changes
